libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1474,11 +1473,18 @@ std::string YulUtilFunctions::cleanUpDynamicByteArrayEndSlotsFunction(ArrayType
 		_args = {"array", "len", "startIndex"};
 		return Whiskers(R"(
 			if gt(len, 31) {
-				let dataArea := <dataLocation>(array)
-				let deleteStart := add(dataArea, <div32Ceil>(startIndex))
-				// If we are clearing array to be short byte array, we want to clear only data starting from array data area.
-				if lt(startIndex, 32) { deleteStart := dataArea }
-				<clearStorageRange>(deleteStart, add(dataArea, <div32Ceil>(len)))
+				if gt(len, startIndex) {
+					let dataArea := <dataLocation>(array)
+					let oldSlotCount := <div32Ceil>(len)
+					let deleteStart := add(dataArea, <div32Ceil>(startIndex))
+					let slotCount := sub(oldSlotCount, <div32Ceil>(startIndex))
+					// If we are clearing array to be short byte array, we want to clear only data starting from array data area.
+					if lt(startIndex, 32) {
+						deleteStart := dataArea
+						slotCount := oldSlotCount
+					}
+					<clearStorageRange>(deleteStart, slotCount)
```

Comment:
I'd do it like this:

```suggestion
					let newSlotCount := <div32Ceil>(startIndex)
					// If we are clearing array to be short byte array, we want to clear only data starting from array data area.
					if lt(startIndex, 32) {
						newSlotCount := 0
					}
					let deleteStart := add(dataArea, newSlotCount)
					<clearStorageRange>(deleteStart, sub(oldSlotCount, newSlotCount))
```

This does not repeat the `<div32Ceil>()` call and does less in the `if`. It also eliminates the `slotCount`, which is too vaguely named and may be a little misleading because of this (in other functions it represents the whole area).

Reply:
Indeed. Thanks :)